### PR TITLE
Deduplicate encode track logic

### PR DESCRIPTION
**Pull request by LordAro**

### CODE REVIEW

File: src/Classes/NIPSWeb/NIPSWeb_ManagedItem.php [hunk 1]
```diff
@@ -227,31 +230,13 @@ public static function storeItem($tmpid, $title)
          * 3- Original file for potential future conversions
          */
         $tmpfile = Config::$audio_upload_tmp_dir.'/'.$tmpid;
-
-        if (!$item->getFolder()) {
-            //Creating folders failed.
-            return ['status' => 'FAIL', 'error' => 'Folders could not be created.', 'fileid' => $_REQUEST['fileid']];
-        }
         $dbfile = $item->getFolder().'/'.$item->getID();
+        CoreUtils::encodeTrack($tmpfile, $dbfile);
```
Comment (DinCahill) [inline src/Classes/NIPSWeb/NIPSWeb_ManagedItem.php]: Why two calls to `encodeTrack`?

Comment (LordAro) [inline src/Classes/NIPSWeb/NIPSWeb_ManagedItem.php]: because i can't read diffs properly

File: src/Classes/MyRadio/CoreUtils.php [hunk 2]
```diff
@@ -233,6 +233,42 @@ public static function makeInterval($start, $end)
...
+    {
+        $commands = [
+            'mp3' => "nice -n 15 ffmpeg -i '{$tmpfile}' -ab 192k -f mp3 -map 0:a '{$dbfile}.mp3'",
+            'ogg' => "nice -n 15 ffmpeg -i '{$tmpfile}' -acodec libvorbis -ab 192k -map 0:a '{$dbfile}.ogg'"
+        ];
+        $escaped_commands = array_map('escapeshellcmd', $commands);
+        $failed_formats = [];
+
+        foreach ($escaped_commands as $format => $command) {
+            exec($command, $command_stdout, $command_exit_code);
```
Comment (LordAro) [inline src/Classes/MyRadio/CoreUtils.php]: where do the stdout/stderr variables come from? where are they used?

**Comment by mstratford** [comment]
#673 Maybe something to fix here? Not sure if we want to open the question to the Track Uploader UI or not? Would be useful to be able to simply replace tracks, but not sure if this should go under the edit form instead?

**Comment by LordAro** [comment]
We could either change the ffmpeg invocation so that it'll overwrite files (`-y`) or return an error if the file already exists

Not sure which is best. Should certainly never happen for database entries, given the name of the file is the primary key, but manageditems might be a bit different...

File: src/Classes/MyRadio/CoreUtils.php (hunk 2 above)
Comment (LloydW93) [inline src/Classes/MyRadio/CoreUtils.php]: http://php.net/manual/en/function.exec.php - these are passed by reference and therefore defined by exec.

**Comment by LloydW93** [comment]
I'm a bit concerned about CoreUtils just becoming a 'stuff' class.

Well, it already is, but we should probably try not to continue this.

**Comment by LordAro** [comment]
It's less of a "stuff" class than it used to be! URLUtils and AuthUtils are things these days. I'm not sure where else it would go, is all.

Alternatively, splitting CoreUtils further? ideas on what should be split out?

**Pull request merged**